runtime/darwinia/src/lib.rs

```diff
@@ -590,28 +592,38 @@ sp_api::impl_runtime_apis! {
 	}
 }
 
-fn migrate() -> Weight {
-	// --- paritytech ---
-	#[allow(unused)]
-	use frame_support::migration;
+pub struct CustomOnRuntimeUpgrade;
+impl OnRuntimeUpgrade for CustomOnRuntimeUpgrade {
+	fn on_runtime_upgrade() -> Weight {
+		// --- paritytech ---
+		use frame_support::traits::PalletInfo;
 
-	// TODO: Move to S2S
-	// const CrabBackingPalletId: PalletId = PalletId(*b"da/crabk");
+		// TODO: Move to S2S
+		// const CrabBackingPalletId: PalletId = PalletId(*b"da/crabk");
 
-	0
-	// RuntimeBlockWeights::get().max_block
-}
+		let name = <Runtime as frame_system::Config>::PalletInfo::name::<Grandpa>()
+			.expect("grandpa is part of pallets in construct_runtime, so it has a name; qed");
```

Comment:
will the expect here be dangerous?

Reply:
https://github.com/paritytech/polkadot/pull/2994#discussion_r730644771

Reply:
> https://github.com/paritytech/polkadot/pull/2994#discussion_r730644771

I agree your opinion on this, both are fine, but if let Some is better than expect.